core/src/main/java/io/camunda/migrator/converter/FlowNodeConverter.java

```diff
@@ -8,57 +8,129 @@
+        .partitionId(C7_HISTORY_PARTITION_ID)
+        .flowNodeName(legacyFlowNode.getActivityName())
+        .flowNodeScopeKey(flowNodeScopeKey)
+        .flowNodeInstanceKey(flowNodeInstanceKey)
+        .flowNodeId(getFlowNodeId(legacyFlowNode))
         .processInstanceKey(processInstanceKey)
         .processDefinitionKey(processDefinitionKey)
-        .processDefinitionId(flowNode.getProcessDefinitionKey())
-        .startDate(convertDate(flowNode.getStartTime()))
-        .endDate(convertDate(flowNode.getEndTime()))
-        .type(convertType(flowNode.getActivityType()))
-        .tenantId(flowNode.getTenantId())
-        .state(null) // TODO: Doesn't exist in C7 activity instance. Inherited from process instance.
-        .treePath(null) // TODO: Doesn't exist in C7 activity instance. Not yet supported by C8 RDBMS
-        .incidentKey(null) // TODO Doesn't exist in C7 activity instance.
-        .numSubprocessIncidents(null) // TODO: increment/decrement when incident exist in subprocess. C8 RDBMS specific.
+        .processDefinitionId(legacyFlowNode.getProcessDefinitionKey())
+        .startDate(convertDate(legacyFlowNode.getStartTime()))
+        .endDate(convertDate(legacyFlowNode.getEndTime()))
+        .type(convertType(legacyFlowNode.getActivityType()))
+        .tenantId(legacyFlowNode.getTenantId())
+        .state(getState(legacyFlowNode))
+        .treePath(buildTreePath(parentTreePath, flowNodeInstanceKey))
+        .incidentKey(null) // will be set when incident is created
+        .hasIncident(false) // will be set when incident is created
+        .numSubprocessIncidents(null)
         .build();
   }
 
+  private String getFlowNodeId(HistoricActivityInstance legacyFlowNode) {
+    return legacyFlowNode.getActivityId().replace(MULTI_INSTANCE_BODY_SUFFIX, "");
+  }
+
+  private String buildTreePath(String parentTreePath, Long flowNodeInstanceKey) {
+    return String.join("/", parentTreePath, (flowNodeInstanceKey.toString()));
+  }
+
+  public void registerIncident(FlowNodeInstanceWriter flowNodeInstanceWriter, FlowNodeInstanceDbModel flowNodeInstance, Long incidentKey) {
+    flowNodeInstanceWriter.createIncident(flowNodeInstance.flowNodeInstanceKey(), incidentKey);
+  }
+
+  public void registerSubprocessIncident(FlowNodeInstanceWriter flowNodeInstanceWriter, FlowNodeInstanceDbModel flowNodeInstance) {
+    flowNodeInstanceWriter.createSubprocessIncident(flowNodeInstance.flowNodeInstanceKey());
+  }
+
+
+  protected FlowNodeInstanceEntity.FlowNodeState getState(HistoricActivityInstance legacyFlowNode) {
+    if (legacyFlowNode.getEndTime() == null) {
+      return FlowNodeInstanceEntity.FlowNodeState.ACTIVE;
+    } else if (legacyFlowNode.isCanceled()) {
+      return FlowNodeInstanceEntity.FlowNodeState.TERMINATED;
+    } else {
+      return FlowNodeInstanceEntity.FlowNodeState.COMPLETED;
+    }
+  }
+
+//  In c7 for example we have event subscriptions
   protected FlowNodeType convertType(String activityType) {
```

Comment:
convertType is still work in progress, will create a follow up PR in the same issue.